src/Streams.Core/Streams.fs

```diff
@@ -4,20 +4,18 @@ open System.Collections.Generic
 open System.Threading
 
 /// Provides on-demand iteration 
-type Iterator = {
+type (* internal *) Iterator = 
     /// Function for on-demand processing
-    TryAdvance : unit -> bool 
+    abstract TryAdvance : unit -> bool 
     /// Cleanup function
-    Dispose : unit -> unit 
-}
+    abstract Dispose : unit -> unit 
 
 /// Provides functions for iteration
-type Iterable = {
+type (* internal *) Iterable = 
```

Comment:
@palladin Some questions on this type Iterable. Looking at the implementation I'm not sure Iterator and Iterable need be separated types. For example the Iterator property (previously a field) needs to hold a single (stateful) Iterator object instance effectively allocated at the same time as Iterable.  So why not make it a single object, e.g.

```
type (* internal *) Iterator = 
    /// Function for bulk processing
    abstract Bulk : unit -> unit 
    /// Iterator for on-demand processing
    abstract TryAdvance : unit -> bool 
    /// Cleanup function
    abstract Dispose : unit -> unit 
```
where there is a contract that either "Bulk" is called once, _or_ TryAdvance is called multiple times?